File: Easy/206. Reverse Linked List/Solution.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(!head or !head->next) return head;
        auto *cur = head->next;
        head->next = nullptr;
        while(cur) {
            auto *go = cur->next;
            cur->next = head;
            head = cur;
            cur = go;
        }
        return head;
    }
};

File: Easy/206. Reverse Linked List/Solution.js
var reverseList = function (head) {
    if (head == null || head.next == null) return head;
    let cur = head.next;
    head.next = null;
    while (cur != null) {
        let go = cur.next;
        cur.next = head;
        head = cur;
        cur = go;
    }
    return head;
};